librad/src/net/quic.rs

```diff
@@ -103,6 +103,19 @@ impl Endpoint {
             None => Ok(mk_connection(peer, conn)),
         }
     }
+
+    /// Close all of this endpoint's connections immediately and cease accepting
+    /// new connections.
+    ///
+    /// Should only ever by called when the protocol stack is shut down.
+    ///
+    /// Morally, this method should consume `self`, but we shut down from a
+    /// `Drop` impl, so all we got is `&mut self`.
+    pub(crate) fn shutdown(&mut self) {
```

Comment:
Do you want to exclusively enforce drop for protoocl shutdown semantics? Or is there a possibility to have this be part of the public interface. Found myself in the situation where in tests I want to spin up a peer have do some operations and then shut it down.

Reply:
There is no other way -- the `quinn::Endpoint` only releases all resources when dropped. So what we could have is to _suspend_ the network, but I'm not sure that makes a lot of sense.

Reply:
Sound. Guess as long as the documentation of the `Protocol` explicitely states the drop semantics it should suffice.